include/brickengine/input.hpp

```diff
@@ -6,30 +6,49 @@
 #include <algorithm>
 #include <tuple>
 #include <utility>
+#include <map>
 
 #include "SDL2/SDL.h"
 #include "brickengine/input_keycode.hpp"
 
 template<typename T>
 class BrickInput {
 public:
-    BrickInput(){
+    BrickInput() {
         generateInputs();
+        if(SDL_GameControllerAddMappingsFromFile("assets/controller/gamecontrollerdb.txt") == -1) {
+            std::cout << "error" << std::endl;
+        }
+        // Checking for joysticks
+        for (int i = 0; i < SDL_NumJoysticks(); ++i) {
+            auto controller = SDL_GameControllerOpen(i);
+            auto id = SDL_JoystickInstanceID(SDL_GameControllerGetJoystick(controller));
+            if(controller) {
+                std::cout << "Found controller: " << SDL_GameControllerNameForIndex(i) << std::endl;
+                controllers[id] = controller;
+            } else {
+                std::cout << "Couldn't open joystick: " << i << ", skipping...";
+            }
```

Comment:
Why isn't this code the same as with the SDL_JOYDEVICEADDED event? Couldn't you here also just loop over the players and add the controller to the player?